pkg/s3select/select.go

```diff
@@ -282,75 +309,97 @@ func (reader *Input) processSelectReq(reqColNames []string, alias string, whereC
+func processColNameIndex(record string, reqColNames []string, columns []string, inputType SelectQuery) (string, error) {
 	row := make([]string, len(reqColNames))
 	for i := 0; i < len(reqColNames); i++ {
 		// COALESCE AND NULLIF do not support index based access.
 		if reqColNames[0] == "0" {
 			return "", ErrInvalidColumnIndex
 		}
-		// Subtract 1 because AWS Indexing is not 0 based, it starts at 1.
 		mytempindex, err := strconv.Atoi(reqColNames[i])
-		if err != nil {
-			return "", ErrMissingHeaders
-		}
-		mytempindex = mytempindex - 1
 		if mytempindex > len(columns) {
 			return "", ErrInvalidColumnIndex
 		}
-		row[i] = record[mytempindex]
+
+		if err != nil {
+			return "", ErrMissingHeaders
+		}
+		// Subtract 1 because AWS Indexing is not 0 based, it starts at 1 generating the key like "_1".
+		row[i] = jsonValue(string("_"+strconv.Itoa(mytempindex-1)), record)
 	}
-	rowStr := strings.Join(row, reader.options.OutputFieldDelimiter)
-	if len(rowStr) > 1000000 {
+	rowStr := strings.Join(row, inputType.getOutputFieldDelimiter())
+	if len(rowStr) > MaxCharsPerRecord {
 		return "", ErrOverMaxRecordSize
 	}
+
 	return rowStr, nil
 }
 
 // processColNameLiteral is the function which creates the row for an name based
 // query.
-func (reader *Input) processColNameLiteral(record []string, reqColNames []string, columns []string, columnsMap map[string]int, myFunc *SelectFuncs) (string, error) {
+func processColNameLiteral(record string, reqColNames []string, myFunc *SelectFuncs, inputType SelectQuery) (string, error) {
 	row := make([]string, len(reqColNames))
 	for i := 0; i < len(reqColNames); i++ {
 		// this is the case to deal with COALESCE.
 		if reqColNames[i] == "" && isValidFunc(myFunc.index, i) {
-			row[i] = evaluateFuncExpr(myFunc.funcExpr[i], "", record, columnsMap)
+			row[i] = evaluateFuncExpr(myFunc.funcExpr[i], "", record)
 			continue
 		}
-		myTempIndex, notFound := columnsMap[trimQuotes(reqColNames[i])]
-		if !notFound {
-			return "", ErrMissingHeaders
-		}
-		row[i] = record[myTempIndex]
+		row[i] = jsonValue(reqColNames[i], record)
+		//myRow = writeRow(myRow, jsonValue(reqColNames[i], record), inputType.getOutputFieldDelimiter(), len(reqColNames))
 	}
-	rowStr := strings.Join(row, reader.options.OutputFieldDelimiter)
-	if len(rowStr) > 1000000 {
+	rowStr := strings.Join(row, inputType.getOutputFieldDelimiter())
+	if len(rowStr) > MaxCharsPerRecord {
 		return "", ErrOverMaxRecordSize
 	}
 	return rowStr, nil
 }
 
-// aggregationFunctions performs the actual aggregation methods on the
-// given row, it uses an array defined for the main parsing function
-// to keep track of values.
-func aggregationFunctions(counter int, filtrCount int, myAggVals []float64, columnsMap map[string]int, storeReqCols []string, storeFunctions []string, record []string) error {
+// aggregationFunctions is a function which performs the actual aggregation
+// methods on the given row, it uses an array defined the the main parsing
```

Comment:
Should `it uses an array defined the the ` be `it uses an array defined in the `

Reply:
Done